### PR TITLE
Add Expense Security Checks

### PR DESCRIPTION
**Pull request by kewitz**
Resolves https://github.com/opencollective/opencollective/issues/5183

### CODE REVIEW

File: server/graphql/common/expenses.ts [hunk 1]
```diff
@@ -209,6 +209,10 @@ export const canSeeExpensePayeeLocation: ExpensePermissionEvaluator = async (req
   return remoteUserMeetsOneCondition(req, expense, [isOwner, isCollectiveAdmin, isCollectiveAccountant, isHostAdmin]);
 };
 
+export const canSeeExpenseSecurityChceks: ExpensePermissionEvaluator = async (req, expense) => {
```
Comment (hdiniz) [inline server/graphql/common/expenses.ts:212]: typo:
```suggestion
export const canSeeExpenseSecurityChecks: ExpensePermissionEvaluator = async (req, expense) => {
```

File: server/graphql/v2/object/Expense.js [hunk 2]
```diff
@@ -373,6 +375,14 @@ const Expense = new GraphQLObjectType({
           return expense.getRecurringExpense();
         },
       },
+      securityChecks: {
+        type: new GraphQLList(SecurityCheck),
+        async resolve(expense, _, req) {
+          if (await ExpenseLib.canSeeExpenseSecurityChceks(req, expense)) {
```
Comment (hdiniz) [inline server/graphql/v2/object/Expense.js:381]:
```suggestion
          if (await ExpenseLib.canSeeExpenseSecurityChecks(req, expense)) {
```

**Review by vaughanj10 — CHANGES_REQUESTED** [review]
This looks really good! Nice job! :smile: 

I have a couple of requests around adding tests for `checkExpense`, and adding some clarifying comments in the code.

File: server/lib/security/expense.ts [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    details: `Expense submitted by ${expense.User.collective.name} (${expense.User.email})`,
+  });
+  addBooleanCheck(!userIsHostAdmin && expense.User.isAdminOfCollective(expense.collective), {
+    scope: Scope.USER,
+    level: Level.LOW,
+    message: `User is the admin of this collective`,
+    details: `Expense submitted by ${expense.User.collective.name} (${expense.User.email})`,
+  });
+
+  const checkStats = async (where, { scope, details }: { scope: Scope; details?: string }) => {
```
[the remark this answers is not in the document]
Comment (kewitz) [reply server/lib/security/expense.ts:103]: I agree. I much rather compose functional expressions.

File: server/lib/security/expense.ts [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+      'status',
+      [sequelize.fn('COUNT', sequelize.col('id')), 'count'],
+      [sequelize.fn('MAX', sequelize.col('createdAt')), 'lastCreatedAt'],
+    ],
+    group: ['status'],
+    order: [['lastCreatedAt', 'desc']],
+    raw: true,
+  });
+
+export const checkExpense = async (expense: typeof models.Expense): Promise<SecurityCheck[]> => {
```
[the remark this answers is not in the document]
Comment (kewitz) [reply server/lib/security/expense.ts:42]: Unfortunately, these tests are not set in stone, and user requests are very subjective.
I'll add a generic test to make sure this works as expected, but I'll ignore any parameters and specifics for now.

**Pull request merged**